compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -843,13 +843,25 @@ impl<'hir> LoweringContext<'_, 'hir> {
         // mut __awaitee => loop { ... }
         let awaitee_arm = self.arm(awaitee_pat, loop_expr);
 
-        // `match ::std::future::IntoFuture::into_future(<expr>) { ... }`
-        let into_future_expr = self.expr_call_lang_item_fn(
-            span,
-            hir::LangItem::IntoFutureIntoFuture,
-            arena_vec![self; expr],
-            Some(expr_hir_id),
+        // `match <expr>.into_future() { ... }`
```

Comment:
this isn't *exactly* the desugaring, so it probably should be a more detailed note explaining the exact semantics here.